pkg/oc/admin/diagnostics/client.go

```diff
@@ -6,21 +6,22 @@ import (
 	"k8s.io/apimachinery/pkg/util/sets"
 	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
 
+	"github.com/openshift/origin/pkg/cmd/util/variable"
 	clientdiags "github.com/openshift/origin/pkg/oc/admin/diagnostics/diagnostics/client"
 	networkdiags "github.com/openshift/origin/pkg/oc/admin/diagnostics/diagnostics/network"
 	"github.com/openshift/origin/pkg/oc/admin/diagnostics/diagnostics/types"
 )
 
-var (
-	// availableClientDiagnostics contains the names of client diagnostics that can be executed
-	// during a single run of diagnostics. Add more diagnostics to the list as they are defined.
-	availableClientDiagnostics = sets.NewString(clientdiags.ConfigContextsName, clientdiags.DiagnosticPodName, networkdiags.NetworkDiagnosticName)
-)
+// availableClientDiagnostics returns definitions of client diagnostics that can be executed
+// during a single run of diagnostics. Add more diagnostics to the list as they are defined.
+func availableClientDiagnostics() types.DiagnosticList {
+	return types.DiagnosticList{clientdiags.ConfigContext{}, &clientdiags.DiagnosticPod{}, &networkdiags.NetworkDiagnostic{}}
+}
 
 // buildClientDiagnostics builds client Diagnostic objects based on the rawConfig passed in.
 // Returns the Diagnostics built, "ok" bool for whether to proceed or abort, and an error if any was encountered during the building of diagnostics.) {
-func (o DiagnosticsOptions) buildClientDiagnostics(rawConfig *clientcmdapi.Config) ([]types.Diagnostic, bool, error) {
```

Comment:
Any strong reason for changing the struct name? `*Options` is the pattern used in most commands.

Reply:
No, actually, just struck me as odd. Didn't realize it was a pattern, happy to change it back.